src/literal.h

```diff
@@ -43,6 +43,11 @@ class Literal {
     Name func;
     // exnref package. `nullptr` indicates a `null` value.
     std::unique_ptr<ExceptionPackage> exn;
+    // A reference to GC data, either a Struct or an Array. For  both of those
+    // we store the referred data as a Literals object (which is natural for an
+    // Array, and for a Struct, is just the fields in order). The type is used
+    // to indicate whether this is a Struct or an Array, and of what type.
+    std::shared_ptr<Literals> gcData;
```

Comment:
Any reason it is not a `unique_ptr` but `shared_ptr`?

Reply:
This is intentional - it's supposed to properly model a reference to GC data on the heap. That is, two Literals can be references to the same data. If the data is modified using one of them, reading from the other reference will show the change, etc.

For function references we have the function name, which works out in the same way.